### PR TITLE
Folder structure

### PR DESCRIPTION
**Pull request by ReneganRonin**
This will be our final folder structure for our bot project. This PR is related to [Issue #10 - Folder structure](https://github.com/gurkult/gurkbot/issues/10). Please review this my fellow cucumbers.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# IDE and Editors
+.vscode/
+.idea/
+
+# Python Stuff
+*.python-version
+*__pycache__/
```
Comment (Xithrius) [inline .gitignore:7]: Not only should we ignore the .env file, but we should ignore everything else that may possibly come in the way.
```suggestion
# Byte-compiled / optimized / DLL files
__pycache__/
*.py[cod]
*$py.class

# C extensions
*.so

# Distribution / packaging
.Python
build/
develop-eggs/
dist/
downloads/
eggs/
.eggs/
lib/
lib64/
parts/
sdist/
var/
wheels/
*.egg-info/
.installed.cfg
*.egg
MANIFEST

# PyInstaller
#  Usually these files are written by a python script from a template
#  before PyInstaller builds the exe, so as to inject date/other infos into it.
*.manifest
*.spec

# Installer logs
pip-log.txt
pip-delete-this-directory.txt

# Unit test / coverage reports
htmlcov/
.tox/
.coverage
.coverage.*
.cache
nosetests.xml
coverage.xml
*.cover
.hypothesis/
.pytest_cache/

# Translations
*.mo
*.pot

# Django stuff:
*.log
local_settings.py
db.sqlite3

# Flask stuff:
instance/
.webassets-cache

# Scrapy stuff:
.scrapy

# Sphinx documentation
docs/_build/

# PyBuilder
target/

# Jupyter Notebook
.ipynb_checkpoints

# pyenv
.python-version

# celery beat schedule file
celerybeat-schedule

# SageMath parsed files
*.sage.py

# Environments
.env
.venv
env/
venv/
ENV/
env.bak/
venv.bak/

# Spyder project settings
.spyderproject
.spyproject

# Rope project settings
.ropeproject

# mkdocs documentation
/site

# mypy
.mypy_cache/

# IDE and Editors
.vscode/
.idea/
```

**Review by Xithrius — APPROVED** [review]
Other than the changes that I'd make to the `.gitignore` file, the PR looks good to me.

**Comment by ReneganRonin** [comment]
I have committed the changes. Waiting for the for the PR to be merged.

**Comment by Xithrius** [comment]
@ReneganRonin one last thing. 

Delete the `.env` file and replace it with `.env-example` so we can have an example file for future people.

After that you'll be good to go.

**Comment by Shivansh-007** [comment]
cool , looks good, just like xith said, change the `.env` file  to either `.env.template` or `.env-example`

**Comment by Vyvy-vi** [comment]
perhaps `DS_Store` could also be added?
(new contributors sometimes commit those)

**Comment by ReneganRonin** [comment]
@Vyvy-vi uhm whats a `.DS_Store`? also @Xithrius i changed it to `.env-template`.

**Comment by Vyvy-vi** [comment]
`.DS_Store` is a file on macos, that holds info about the folder's properties(how icons are arranged etc..)

**Comment by Xithrius** [comment]
> `.DS_Store` is a file on macos, that holds info about the folder's properties(how icons are arranged etc..)

@ReneganRonin what @Vyvy-vi meant by this was to add the `.DS_Store` file to the `.gitignore` file. There is no need to create another file.

**Comment by Vyvy-vi** [comment]
Oh, I should have been more specific 😞 
Sorry!

**Comment by ReneganRonin** [comment]
it is done. sorry for the confusion. @Xithrius @Vyvy-vi

**Comment by Xithrius** [comment]
Everything looks good to me! Nice work.

**Comment by ReneganRonin** [comment]
@fisher60 should we create a new branch called "core"?

**Comment by Vyvy-vi** [comment]
Isn't `main` analogous to `core`?
Imho, we should make a new `develop` branch or make a new `core` branch and let `main` act as the dev-branch 🤔

**Pull request merged**